_source/_posts/2022-01-20-vue-applications-nuxt.md

```diff
@@ -0,0 +1,357 @@
+        <b-navbar-nav>
+          <b-nav-item href="/dashboard"> Dashboard </b-nav-item>
+        </b-navbar-nav>
+
+        <b-navbar-nav v-if="loggedIn" class="ml-auto">
+          <b-button @click="logout" size="sm" class="my-2 my-sm-0" type="submit"
+            >Logout
+          </b-button>
+        </b-navbar-nav>
+
+        <b-navbar-nav v-else class="ml-auto">
+          <b-button @click="login" size="sm" class="my-2 my-sm-0" type="submit"
+            >Login
+          </b-button>
+        </b-navbar-nav>
+
+      </b-navbar>
+
+      <nuxt />
+
+      <footer id="sticky-footer" class="py-4 bg-dark text-white-50">
+        <div class="fluid-container footer">
+          <small>Copyright &copy;{{ year }} </small>
+          <br />
+
+          A small app built with
+          <a href="https://nuxtjs.org/" target="blank">Nuxt</a>, Protected by
+          <a href="https://www.okta.com/" target="blank">Okta</a>, Written by
+          <a href="https://profile.fishbowlllc.com" target="blank">Nik Fisher</a
+          >.
+        </div>
+      </footer>
+    </b-container>
+  </div>
+</template>
+
+<script>
+export default {
+  data() {
+    return {
+      loggedIn: this.$auth.$state.loggedIn,
+      year: new Date().getFullYear(),
+    }
+  },
+  methods: {
+    logout() {
+      this.$auth.logout();
+    },
+    login() {
+      this.$auth.loginWith('okta').then(result => window.location = "/Dashboard");
+    }
+  },
+}
+</script>
+
+<style scoped>
+.fluid-container.footer > *:last-child {
+  margin-bottom: 0px;
+  color: #fff;
+}
+</style>
+```
+{% endraw %}
+
+You will also need a basic landing page that isn't under authentication. Your landing page will give some information about the application. It will also contain a redirect for authenticated users to route them to the `Dashboard` page. 
+
+Open the `index.vue` page in the `pages` directory and replace the code with the following.
+
+```html
+<template>
+  <div id="page-content">
+    <b-jumbotron
+      style="margin-top: 5vh"
+      header="Lets Get Some Posts"
+      lead="A Simple App with Nuxt and Okta"
+    >
+      <p>For more information visit their websites</p>
+      <b-button variant="primary" href="https://nuxtjs.org/" target="blank"
+        >Nuxt</b-button
+      >
+      <b-button
+        variant="outline-primary"
+        href="https://www.okta.com/"
+        target="blank"
+        >Okta</b-button
+      >
+    </b-jumbotron>
+  </div>
+</template>
+
+<script>
+export default {
+  beforeMount() {
+    if (this.$auth.$state.loggedIn) window.location = '/Dashboard'
+  },
+}
+</script>
+```
+
+Next, you can add the `Dashboard` page. This page makes use of the `data()` webhook to get some data from a server using Axios. This data is then displayed in a table. This page also makes use of the `auth` middleware to enforce authentication on this page. When a user attempts to hit this page they will be rerouted to your login page if they are not authenticated.
```

Comment:
```suggestion
Next, you can add the `Dashboard` page. This page makes use of the `data()` webhook to get some data from a server using Axios, which it then displays in a table. The `Dashboard` page also makes use of the `auth` middleware to enforce authentication on this page. When a user attempts to hit this page, they will be rerouted to your login page if they are not authenticated.
```
routes around the repetition of "this" and "this page" to open several sentences. increases readability.